### PR TITLE
Refactor public event page to use tickets API

### LINKED ISSUE #460: Implement tickets API for public/index
**I'm submitting a ...**  (check one with "x")
- [ ] bug report
- [x] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-frontend

Implement tickets API for the public/index route

Reference: https://open-event-api.herokuapp.com/#tickets

### PR DESCRIPTION
**Pull request by CosmicCoder96**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Refactorw public event page to use tickets API

#### Changes proposed in this pull request:
Remlove dummy data for tickets and use API

Fixes #460

### CODE REVIEW

File: app/routes/public/index.js [hunk 1]
```diff
@@ -5,21 +5,17 @@ const { Route, RSVP } = Ember;
 export default Route.extend({
   model() {
     return RSVP.hash({
-      event: this._super(...arguments),
-
-      speakers: [{  name: 'Speaker 1', organisation: 'Org 1', isFeatured: false, socialLinks: [{ name: 'facebook', url: '#' }]  },
+      event    : this._super(...arguments),
+      tickets  : this._super(...arguments).get('tickets'),
```
Comment (CosmicCoder96) [inline app/routes/public/index.js]: @niranjan94 Please guide regarding filtering server side.

**Comment by CosmicCoder96** [comment]
@niranjan94 @geekyd Please review, have added the new module, it's taking care of passing on the filter queries backend side.

File: app/components/public/ticket-list.js [hunk 2]
```diff
@@ -1,20 +1,27 @@
...
 import FormMixin from 'open-event-frontend/mixins/form';
-import { sumBy } from 'lodash';
 
 const { Component, computed } = Ember;
 
 export default Component.extend(FormMixin, {
   classNames: ['ui', 'segments', 'ticket-list'],
 
   hasTicketsInOrder: computed('tickets.@each.orderQuantity', function() {
-    return sumBy(this.get('tickets'), 'orderQuantity') > 0;
```
Comment (niranjan94) [inline app/components/public/ticket-list.js]: You can still use this right ? 

```js
return sumBy(this.get('tickets'), ticket => ticket.getWithDefault('orderQuantity', 0));
```

File: app/components/public/ticket-list.js [hunk 3]
```diff
@@ -1,20 +1,27 @@
...
+      if (ticket.orderQuantity) {
+        totalQuantity += ticket.orderQuantity;
+      }
+    });
+    return totalQuantity > 0;
   }),
 
   total: computed('tickets.@each.orderQuantity', function() {
     let sum = 0.0;
     this.get('tickets').forEach(ticket => {
```
Comment (niranjan94) [inline app/components/public/ticket-list.js]: This could be written as 

```js
return sumBy(
  this.get('tickets'), 
  ticket => ticket.get('price') * ticket.getWithDefault('orderQuantity', 0)
);
```

File: app/routes/public/index.js [hunk 4]
```diff
@@ -1,12 +1,30 @@
...
+import moment from 'moment';
 
 const { Route, RSVP } = Ember;
 
 export default Route.extend({
   model() {
     return RSVP.hash({
-      event: this._super(...arguments),
-
+      event   : this._super(...arguments),
```
Comment (niranjan94) [inline app/routes/public/index.js]: You're calling the super method twice. You should only call once. Store the result in a variable and use it twice.

File: app/components/public/ticket-list.js (hunk 2 above)
Comment (CosmicCoder96) [inline app/components/public/ticket-list.js]: @niranjan94 well.. it doesn't seem to be  working throws a  *cannot  use get for undefined* error

Comment (CosmicCoder96) [inline app/components/public/ticket-list.js]: @niranjan94 
I think it is an issue with this.get('tickets') being an object, if I use `this.get('tickets').toArray()` it works nicely. I am changing it to that along with the toArray(), please let me know of a better solution for it, if required

Comment (niranjan94) [inline app/components/public/ticket-list.js]: Cool. Use `toArray`

File: tests/integration/components/public/ticket-list-test.js [hunk 5]
```diff
@@ -1,14 +1,50 @@
 import { test } from 'ember-qunit';
+import Ember from 'ember';
 import moduleForComponent from 'open-event-frontend/tests/helpers/component-helper';
 import hbs from 'htmlbars-inline-precompile';
 
-moduleForComponent('public/ticket-list', 'Integration | Component | public/ticket list');
+const { Object, ArrayProxy } = Ember;
```
Comment (niranjan94) [inline tests/integration/components/public/ticket-list-test.js]: No need array proxy. You can just use `A`
```js
const { Object, A } = Ember;
```

And also, alias `Object` to `EmberObject` to avoid confusion with the native js `Object`

```js
const { Object: EmberObject, A } = Ember;
```

File: app/components/public/ticket-list.js (hunk 2 above)
Comment (CosmicCoder96) [inline app/components/public/ticket-list.js]: @niranjan94  umm.. do you mean to say, cool do what have I done, or use the lodash `toArray` 😅

Comment (niranjan94) [inline app/components/public/ticket-list.js]: `this.get('tickets').toArray()` is cool ...!

File: tests/integration/components/public/ticket-list-test.js (hunk 5 above)
Comment (CosmicCoder96) [inline tests/integration/components/public/ticket-list-test.js]: @niranjan94 A.create fails 3  the tests, the component and the moment and a global failure.
Am I missing something? I just replaced ArrayProxy with A.

Comment (niranjan94) [inline tests/integration/components/public/ticket-list-test.js]: You don't need to put `A.create`. You can just wrap the array in A ... Like 

```
A([])
```

**Comment by CosmicCoder96** [comment]
@niranjan94 Made the changes, please review

File: tests/integration/components/public/ticket-list-test.js [hunk 6]
```diff
@@ -1,14 +1,49 @@
...
 
-const tickets = [{ description: 'Community ticket', date: 'Mon, May 22', price: 2.50, name: 'Community Ticket', type: 'paid', id: 1, quantity: 10, orderQuantity: 0, min: 0, max: 5 },
-  { description: 'Standard ticket', date: 'Mon, May 22', price: 3.05, name: 'Standard Ticket', type: 'paid', id: 2, quantity: 10, orderQuantity: 0, min: 0, max: 5 },
-  { description: 'Super ticket', date: 'Mon, May 22', price: 5.00, name: 'Super Ticket', type: 'paid', id: 3, quantity: 10, orderQuantity: 0, min: 0, max: 5 }
-];
+moduleForComponent('public/ticket-list', 'Integration | Component | public/ticket list');
 
+const tickets =  A(
+  [
+    EmberObject.create({
```
Comment (niranjan94) [inline tests/integration/components/public/ticket-list-test.js]: Instead of ember object, could you create instances of the ticket model itself ?

**Comment by CosmicCoder96** [comment]
> We also need to consider the case where the organiser is using an external link for tickets.
@niranjan94  can this be done in a separate issue?  and if not How to proceed with that?

**Comment by geekyd** [comment]
> We also need to consider the case where the organiser is using an external link for tickets.

@CosmicCoder96 @niranjan94 This should be done in this PR only as it is a part of ticket API integration. All we need to do is check If `is-ticketing-enabled` is set or not & show or hide the ticketing section accordingly.

**Comment by CosmicCoder96** [comment]
@geekyd Yes, I will Implement it, was under the impression that the ticket URL is not part of the event model :)

**Comment by CosmicCoder96** [comment]
@geekyd Done, 
image
@niranjan94 Didn't change the tests as per discussion
Please review again. 
https://cc96-opev-frontend.herokuapp.com/e/0d8858bf

File: app/templates/components/public/ticket-list.hbs [hunk 7]
```diff
@@ -2,109 +2,127 @@
...
-          <th>{{t 'Type'}}</th>
-          <th class="four wide">{{t 'Sales Ends'}}</th>
-          <th class="ui two wide">{{t 'Price'}}</th>
-          <th class="one wide">{{t 'Quantity'}}</th>
-          <th class="ui right aligned two wide">{{t 'Subtotal'}}</th>
-        </tr>
-      </thead>
-      <tbody>
-        {{#each tickets as |ticket|}}
+  {{#if isTicketingEnabled}}
```
Comment (niranjan94) [inline app/templates/components/public/ticket-list.hbs]: This shouldn't be done here. `ticket-list` component is specifically for showing a list of tickets. Nothing else. Do the check and use this component only when there are tickets available. Revert this change from this file please.

Comment (CosmicCoder96) [inline app/templates/components/public/ticket-list.hbs]: @niranjan94 Got it, and btw should the order tickets segment in case of no tickets be in a separate component, or directly in the route template?

Comment (niranjan94) [inline app/templates/components/public/ticket-list.hbs]: Not much is there. Could be in the template.

**Comment by CosmicCoder96** [comment]
@niranjan94 done
